optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/PlannerBenchmarkRunner.java

```diff
@@ -273,7 +258,7 @@ protected void runSingleBenchmarks() {
             }
         }
         // Wait for the benchmarks to complete
-        for (Map.Entry<SubSingleBenchmarkRunner, Future<SubSingleBenchmarkRunner>> futureEntry : futureMap.entrySet()) {
+        futureMap.entrySet().forEach(futureEntry -> {
```

Comment:
These kind of changes are bloating the PR a bit for me, making it hard to grok.
Although I agree we should start using the diamond operator, etc - and through our entire codebase - it rather deal with that in isolation (and preferably after 6.4.0.Final is out to avoid backporting merge conflicts).

Specifically, replacing a for(each) with a .forEach() is a debatable change as it's probably slower (use of streams and lamba's without need for parallel()).